dunamai/__init__.py

```diff
@@ -317,7 +318,44 @@ def __init__(
         :param commit: Commit hash/identifier.
         :param dirty: True if the working directory does not match the commit.
         :param epoch: Optional PEP 440 epoch.
+        :param pattern: Regular expression matched against the base.
+            Refer to `from_any_vcs` for more info.
```

Comment:
Could you explain the use case for this? I didn't really think anyone would need to turn a version string back into a `Version` instance.

If we do add it, I'd probably prefer it to be a separate function, like `def parse_version(version: str) -> Version`.

Reply:
It's use full when you want to modify a version that is gathered from `importlib.metadata`. Since `dunamai` just grabs the string, everything is put inside `base` .

Reply:
Ah, that's a great point.

Okay, let's separate this from the constructor and put it as `Version.parse(cls, version: str, pattern: str = _VERSION_PATTERN) -> "Version"`. That way, the constructor stays simple and predictable.

Reply:
I've pushed an possible implementation. `Version.parse` allows a version as well so that you can use the function as follows: `dunamai.Version.parse(dunamai.get_version("dunamai", third_choice=dunamai.Version.from_any_vcs))`

Reply:
To keep `parse()` simpler, I think I'll just add a `parser` argument to `get_version()` after the merge. `parser=Version` would keep the existing behavior, and `parser=Version.parse` would enable the new behavior.